pkg/ctl/get/nodegroup.go

```diff
@@ -90,20 +90,21 @@ func doGetNodeGroup(cmd *cmdutils.Cmd, ng *api.NodeGroup, params *getCmdParams)
 		summaries = append(summaries, summary)
 	}
 
-	// Empty summary implies no nodegroups
-	if len(summaries) == 0 {
-		if ng.Name == "" {
-			return errors.Errorf("No nodegroups found")
-		}
-		return errors.Errorf("Nodegroup with name %v not found", ng.Name)
-	}
-
 	printer, err := printers.NewPrinter(params.output)
 	if err != nil {
 		return err
 	}
 
 	if params.output == "table" {
+		// Empty summary implies no nodegroups
+		// We only error if the output is table, since if the output
+		// is yaml or json we should return an empty object.
+		if len(summaries) == 0 {
+			if ng.Name == "" {
+				return errors.Errorf("No nodegroups found")
+			}
+			return errors.Errorf("Nodegroup with name %v not found", ng.Name)
```

Comment:
Super-minor nit: 
```suggestion
			return errors.Errorf("nodegroup with name %v not found", ng.Name)
```

(I know this is taken from the surrounding block)